### PR TITLE
Integrate latest Plutarch and haskell-nix changes

### PR DESCRIPTION
**Pull request by bladyjoker**
DONE
- [x] Used extra-hackages
- [x] Used Plutarch hls
- [x] Used explicit index-state
- [x] Recreated the lock file as it was borked in mysterious ways

### CODE REVIEW

**Comment by bladyjoker** [comment]
Dumping issues here...

```
protolude-lib-protolude> --prefix=/nix/store/in2mlhnigvh5pdpzcsh85i05bppj3hk7-protolude-lib-protolude-0.3.2 lib:protolude --package-db=clear --package-db=/nix/store/agshwynlqcpl9aswm2rfa99vc51amykm-protolude-lib-protolude-0.3.2-config/lib/ghc-9.2.3/package.conf.d --exact-configuration --dependency=async=async-2.2.4-CCZgFYbGl2GGn1PgvKLmC5 --dependency=hashable=hashable-1.4.0.2-HuHqB79XQMV4Ee6XIG91Qq --dependency=mtl-compat=mtl-compat-0.2.2-3bZ9KHvOKEkGXEixTMsgLy --dependency=transformers-compat=transformers-compat-0.7.2-AFX8xJOEGOK8jE4fBAbpzG --dependency=Cabal=Cabal-3.6.3.0 --dependency=array=array-0.5.4.0 --dependency=base=base-4.16.2.0 --dependency=binary=binary-0.8.9.0 --dependency=bytestring=bytestring-0.11.3.1 --dependency=containers=containers-0.6.5.1 --dependency=deepseq=deepseq-1.4.6.1 --dependency=directory=directory-1.3.6.2 --dependency=exceptions=exceptions-0.10.4 --dependency=filepath=filepath-1.4.2.2 --dependency=ghc=ghc-9.2.3 --dependency=ghc-bignum=ghc-bignum-1.2 --dependency=ghc-boot=ghc-boot-9.2.3 --dependency=ghc-boot-th=ghc-boot-th-9.2.3 --dependency=ghc-compact=ghc-compact-0.1.0.0 --dependency=ghc-heap=ghc-heap-9.2.3 --dependency=ghc-prim=ghc-prim-0.8.0 --dependency=hpc=hpc-0.6.1.0 --dependency=integer-gmp=integer-gmp-1.1 --dependency=mtl=mtl-2.2.2 --dependency=parsec=parsec-3.1.15.0 --dependency=pretty=pretty-1.1.3.6 --dependency=process=process-1.6.13.2 --dependency=rts=rts --dependency=stm=stm-2.5.0.2 --dependency=template-haskell=template-haskell-2.18.0.0 --dependency=terminfo=terminfo-0.4.1.5 --dependency=text=text-1.2.5.0 --dependency=time=time-1.11.1.1 --dependency=transformers=transformers-0.5.6.2 --dependency=unix=unix-2.7.2.2 --dependency=xhtml=xhtml-3000.2.2.1 --with-ghc=ghc --with-ghc-pkg=ghc-pkg --with-hsc2hs=hsc2hs --with-gcc=cc --with-ld=ld.gold --ghc-option=-optl-fuse-ld=gold --ld-option=-fuse-ld=gold --with-ar=ar --with-strip=strip --disable-executable-stripping --disable-library-stripping --disable-library-profiling --disable-profiling --enable-static --enable-shared --disable-coverage --enable-library-for-ghci --datadir=/nix/store/b387r941hdmyanwsa2ancvw5ldq7kj32-protolude-lib-protolude-0.3.2-data/share/ghc-9.2.3 --enable-split-sections
protolude-lib-protolude> Configuring library for protolude-0.3.1..
protolude-lib-protolude> Setup: '--exact-configuration' was given, but the following flags were not
protolude-lib-protolude> specified: FlagName "dev"

```

**Comment by TotallyNotChase** [comment]
@bladyjoker Does this branch build hassle free in downstream projects?

**Comment by bladyjoker** [comment]
> @bladyjoker Does this branch build hassle free in downstream projects?

It did at [d92fd35](https://github.com/mlabs-haskell/ply/pull/7/commits/d92fd3504096585bfe215dbc0b830054ad3b2f2e) let me try with your additions...

**Comment by TotallyNotChase** [comment]
I tried to add extra hackages to the core stuff :D

File: flake.nix
Comment (bladyjoker) [inline flake.nix:15]: @TotallyNotChase should we keep the rev or just say staging?

File: flake.nix
Comment (bladyjoker) [inline flake.nix:108]: Oh the ergonomics gains are real

File: flake.nix
Comment (bladyjoker) [inline flake.nix:236]: @TotallyNotChase see here I break the nix pattern by just using plutarch.inputs and not orchestrating that at the Flake inputs. You ok with that?

**Comment by bladyjoker** [comment]
> I tried to add extra hackages to the core stuff :D

It works for me!

File: flake.nix
Comment (TotallyNotChase) [reply flake.nix:108]: revolutionary technology really, a whole 500 more lines compared to just using cabal!

File: flake.nix
Comment (TotallyNotChase) [reply flake.nix:236]: As long as it works, it can stay

File: flake.nix
Comment (TotallyNotChase) [reply flake.nix:15]: I'll try with staging actually

**Review by TotallyNotChase — APPROVED** [review]

**Comment by bladyjoker** [comment]
@TotallyNotChase let's wait for the green light and then merge?

**Comment by bladyjoker** [comment]
> Is hydra dead

https://ci.staging.mlabs.city/jobset/mlabs-haskell/ply__pr-7#tabs-evaluations

wth

`Evaluation running since: | 1h ago`

File: flake.nix
Comment (bladyjoker) [inline flake.nix:5]: @TotallyNotChase it might be this!

Comment (TotallyNotChase) [reply flake.nix:5]: we're supposed to use official haskell nix now - I think this should still pass fine (the previous commit with this passed)

Comment (bladyjoker) [reply flake.nix:5]:
```
bladyjoker@morphism:~/Desktop/ply$ nix flake metadata github:input-output-hk/haskell.nix/e3397e0ace4aeadefe941241bd5b7bfefad850a9
Resolved URL:  github:input-output-hk/haskell.nix/e3397e0ace4aeadefe941241bd5b7bfefad850a9
Locked URL:    github:input-output-hk/haskell.nix/e3397e0ace4aeadefe941241bd5b7bfefad850a9
Description:   Alternative Haskell Infrastructure for Nixpkgs
Path:          /nix/store/k35qjhl7mfl0bnq6iaj22rhcpz3rbdki-source
Revision:      e3397e0ace4aeadefe941241bd5b7bfefad850a9
Last modified: 2022-07-26 23:10:13
```

Yeah it's likely rebuilding the world because of this but who knows :shrug:

Comment (TotallyNotChase) [reply flake.nix:5]: screw it, yolo

**Pull request merged**
